components/centraldashboard/public/components/notebook-default-card.js

```diff
@@ -0,0 +1,119 @@
+import '@polymer/paper-toast/paper-toast.js';
+import '@polymer/paper-ripple/paper-ripple.js';
+import '@polymer/paper-item/paper-icon-item.js';
+import '@polymer/paper-icon-button/paper-icon-button.js';
+
+import {html, PolymerElement} from '@polymer/polymer';
+
+import './card-styles.js';
+import css from './notebook-default-card.css';
+import template from './notebook-default-card.pug';
+// eslint-disable-next-line max-len
+import {AppLocalizeBehavior} from '@polymer/app-localize-behavior/app-localize-behavior.js';
+import {mixinBehaviors} from '@polymer/polymer/lib/legacy/class.js';
+
+// eslint-disable-next-line max-len
+export class NotebookDefaultCard
+    extends mixinBehaviors([AppLocalizeBehavior], PolymerElement) {
+    static get template() {
+        return html([`
+            <style>${css.toString()}</style>
+            ${template()}
+        `]);
+    }
+
+    /**
+     * Object describing property-related metadata used by Polymer features
+     */
+    static get properties() {
+        return {
+            defaultNotebookError: Object,
+            namespace: String,
+            loading: {type: Boolean, value: false},
+        };
+    }
+
+    // Functions to render the HTML template ifs correctly
+    _isNotebookUndefined(t) {
+        return t.notebook === undefined;
+    }
+
+    _isNotebookEmpty(t) {
+        return t.notebook.name == '';
+    }
+
+    _isNotebookReady(t) {
+        return t.notebook.status.phase == 'ready';
+    }
+    _isCreateDisabled() {
+        return this.loading;
+    }
+
+
+    _connectNotebook() {
+        // eslint-disable-next-line max-len
+        window.open(`/notebook/${this.namespace}/${this.defaultNotebook.notebook.name}/`);
+    }
+
+    _detailNotebook() {
+        // Possibly need the language flag beforehand and or the jupyter
+        // eslint-disable-next-line max-len
+        window.open(`/notebook/details/${this.namespace}/${this.defaultNotebook.notebook.name}/`);
+    }
+
+    /**
+     * Main ready method for Polymer Elements.
+     */
+    ready() {
+        super.ready();
+    }
+
+    async _createDefaultNotebook() {
+        // Disable the create button
+        this.loading = true;
+        // Create the default notebook
+        const APICreateDefault = this.$.CreateDefaultNotebook;
+        await APICreateDefault.generateRequest().completes.catch((e) => e);
+    }
+
+    /**
+     * Takes an event from iron-ajax and isolates the error from a request that
+     * failed
+     * @param {IronAjaxEvent} e
+     * @return {string}
+     */
+    _isolateErrorFromIronRequest(e) {
+        const bd = e.detail.request.response||{};
+        return bd.error || e.detail.error || e.detail;
+    }
+
+    /**
+     * Iron-Ajax response for creating new Default Notebook
+     * @param {IronAjaxEvent} e
+     */
+    handleNotebookCreate(e) {
+        this.defaultNotebook = e.detail.response.notebook;
+        this.defaultNotebookError = '';
```

Comment:
defaultNotebookError is defined as an object above. Shouldn't this set it to null or something? and not an empty string.